[workers] routes revision (#2739)
cloudflare/cloudflare-docs: products/workers/src/content/platform/routes.md

For zones proxied on Cloudflare\*, route patterns decide what (if any) script is matched based on the URL of that request. Requests are routed through a Workers script when the URL matches a route pattern assigned to that script.

Route patterns can be added with the Cloudflare API or in **Account Home** > [**Workers**](https://dash.cloudflare.com/?zone=workers) > **your Worker** > **Triggers** > **Add route** in the Cloudflare dashboard.

![Workers Route Modal](./add-route-modal.png)

Cloudflare Site routes are comprised of:

- Route URL (refer to [Matching Behavior](#matching-behavior))
- Worker script to execute on matching requests
- Failure mode for rate-limited accounts on the Free plan (see [daily request limits](/platform/limits#request-limits))

The Routes REST API documentation can be found [in the Workers API documentation](https://api.cloudflare.com/#worker-routes-properties).

> \* A zone that you have registered with some registrar (not a `*.workers.dev` subdomain) and set up Cloudflare to serve as [a reverse proxy](https://www.cloudflare.com/learning/cdn/glossary/reverse-proxy/).

Author: @lukeed I feel like we left this in here for a good reason but I am now confused at how this sentence is floating with the \* in the beginning. Is this supposed to match with line 14?

Reviewer: I don't think I've ever seen this – if we did then the workers.dev would have already been formatted

> \* A zone that you have registered with some registrar (not a `*.workers.dev` subdomain) and set up Cloudflare to serve as [a reverse proxy](https://www.cloudflare.com/learning/cdn/glossary/reverse-proxy/).
> 

Author: ```suggestion
```

> - The only supported operator is wildcard `*` which matches zero or more of any character.

Reviewer 2: ```suggestion
- The only supported operator is the wildcard (`*`), which matches zero or more of any character.
```

> - Route patterns may not contain infix wildcards or query parameters, for example, neither `example.com/*.jpg` nor `example.com/?foo=*` are valid route patterns.

Reviewer 2: ```suggestion
- Route patterns may not contain infix wildcards or query parameters. For example, neither `example.com/*.jpg` nor `example.com/?foo=*` are valid route patterns.
```

### Validity

> The set of rules governing route pattern validity:

Reviewer 2: should this be a complete sentence? e.g., `Route pattern validity is governed by the following set of rules:` (excuse the passive)

> The set of rules governing route pattern validity:

Author: ```suggestion
The following set of rules govern route pattern validity.
```

#### Route patterns must include your zone

If your zone is `example.com`, then the simplest possible route pattern you can have is `example.com`, which would match `http://example.com/` and `https://example.com/`, and nothing else. As with a URL, there is an implied path of `/` if you do not specify one.

#### Route patterns may not contain any query parameters

For example, `https://example.com/?anything` is not a valid route pattern.

#### Route patterns may optionally begin with http:// or https://

If you omit a scheme in your route pattern, it will match both `http://` and `https://` URLs. If you include `http://` or `https://`, it will only match HTTP or HTTPS requests, respectively.

- `https://*.example.com/` matches `https://www.example.com/` but not `http://www.example.com/`.

- `*.example.com/` matches both `https://www.example.com/` and `http://www.example.com/`.

#### Hostnames may optionally begin with `*`

> If a route pattern hostname begins with `*`, then it matches the host and all subhosts. If a route pattern hostname begins with `*.`, then it matches only all subhosts.

Reviewer 2: ```suggestion
If a route pattern hostname begins with `*`, then it matches the host and all subhosts. If a route pattern hostname begins with `*.`, then it only matches all subhosts.
```

- `*example.com/` matches `https://example.com/` and `https://www.example.com/`.

- `*.example.com/` matches `https://www.example.com/` but not `https://example.com/`.

#### Paths may optionally end with `*`

If a route pattern path ends with `*`, then it matches all suffixes of that path.

- `https://example.com/path*` matches `https://example.com/path` and `https://example.com/path2` and `https://example.com/path/readme.txt`

<Aside type="warning">

> There is a well-known bug associated with path matching concerning wildcards (`*`) and forward slashes (`/`) that is documented in [Known issues.](/platform/known-issues)

Reviewer 2: ```suggestion
There is a well-known bug associated with path matching concerning wildcards (`*`) and forward slashes (`/`) that is documented in [Known issues](/platform/known-issues).
```
